xtable-core/src/test/java/org/apache/xtable/iceberg/TestIcebergDataHelper.java

```diff
@@ -99,7 +99,9 @@ public class TestIcebergDataHelper {
           NestedField.optional(
               28, "timestamp_micros_nullable_field", Types.TimestampType.withZone()),
           NestedField.optional(
-              30, "timestamp_local_micros_nullable_field", Types.TimestampType.withoutZone()));
+              30, "timestamp_local_micros_nullable_field", Types.TimestampType.withoutZone()),
+          NestedField.optional(33, "uuid_field", Types.UUIDType.get())
```

Comment:
Hi @the-other-tim-brown ,After adding the new line to `COMMON_FIELDS`, I encountered an error when converting from Iceberg to Hudi during the `ITConversionController` test. I’m wondering if this issue might be due to a missing configuration or if there is something specific I should be aware of in this context.

If you could provide some guidance or context on this, it would be greatly appreciated.

```

org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 134.0 failed 1 times, most recent failure: Lost task 0.0 in stage 134.0 (TID 571) (10.0.0.119 executor driver): org.apache.spark.sql.AnalysisException: Illegal Parquet type: FIXED_LEN_BYTE_ARRAY (UUID).
	at org.apache.spark.sql.errors.QueryCompilationErrors$.illegalParquetTypeError(QueryCompilationErrors.scala:1762)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.illegalType$1(ParquetSchemaConverter.scala:206)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.$anonfun$convertPrimitiveField$2(ParquetSchemaConverter.scala:310)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convertPrimitiveField(ParquetSchemaConverter.scala:224)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convertField(ParquetSchemaConverter.scala:187)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.$anonfun$convertInternal$3(ParquetSchemaConverter.scala:147)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.$anonfun$convertInternal$3$adapted(ParquetSchemaConverter.scala:117)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.immutable.Range.foreach(Range.scala:158)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convertInternal(ParquetSchemaConverter.scala:117)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convert(ParquetSchemaConverter.scala:87)
	at org.apache.spark.sql.execution.datasources.parquet.HoodieParquetFileFormatHelper$.buildImplicitSchemaChangeInfo(HoodieParquetFileFormatHelper.scala:31)
	at org.apache.spark.sql.execution.datasources.parquet.Spark34LegacyHoodieParquetFileFormat.$anonfun$buildReaderWithPartitionValues$2(Spark34LegacyHoodieParquetFileFormat.scala:254)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:231)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:293)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:125)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:101)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:53)
	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
	at org.apache.spark.scheduler.Task.run(Task.scala:139)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

Reply:
Since this issue only occurs during the Iceberg to Hudi conversion, while other conversions work fine, I suspect I may have missed some changes in the corresponding converters.

Reply:
It doesn't work as well:
` NestedField.optional(33, "uuid_field", Types.FixedType.ofLength(16))`

But if we add this field in this way, it works:

`NestedField.optional(33, "uuid_field", Types.BinaryType.get())`

Reply:
Does Iceberg to Delta work as expected? If only Hudi is broken, we can list that as a limitation and make a test case for Iceberg to Delta for now. 

Thank you for digging in and providing the context up front of what you have tested and where the issue is, less homework for me :)

Reply:
> Does Iceberg to Delta work as expected? If only Hudi is broken, we can list that as a limitation and make a test case for Iceberg to Delta for now.

The conversion from Iceberg to Delta success, but there is a diff in the result:
- Iceberg: "uuid_field":"8135e4c4-ea58-4ace-a819-4f835cb2b721"
- Delta: "uuid_field":"gTXkxOpYSs6oGU+DXLK3IQ=="

In fact these two data are the same(`8135e4c4-ea58-4ace-a819-4f835cb2b721` equals `8135e4c4-ea58-4ace-a819-4f835cb2b721`), but assertion fails in this diff because the data format in two table is different - data format in the two tables is different—Iceberg supports UUIDs directly, while Delta represents them as binary.

Should we consider adding special comparison logic in `checkDatasetEquivalence` for these specific cases?

Reply:
> Thank you for digging in and providing the context up front of what you have tested and where the issue is, less homework for me :)

My pleasure :) @the-other-tim-brown , this is a great opportunity for me to learn more about Spark and Parquet. Additionally, if we determine that this issue is indeed caused by incomplete support for UUIDs in the Spark Parquet converter, we could potentially contribute some code to the Spark community.

Reply:
I think we can add in a special case on the assertion for these UUIDs. Are you thinking of just adding a transform on the field in the dataset?

Reply:
> Does Iceberg to Delta work as expected? If only Hudi is broken, we can list that as a limitation and make a test case for Iceberg to Delta for now.
> 
> I think we can add in a special case on the assertion for these UUIDs. Are you thinking of just adding a transform on the field in the dataset?

Thanks for the feedback @the-other-tim-brown , for the latest commit, I have completed testing for UUID:
1. Add a UUID test case for Iceberg -> Delta for now 
2. Add special case on assertion by processing binary UUID into string before comparison
3. Make "xtableLogicalType" into a constant